src/ng/q.js

```diff
@@ -276,7 +315,7 @@ function qFactory(nextTick, exceptionHandler) {
 
       this.$$state.pending = this.$$state.pending || [];
       this.$$state.pending.push([result, onFulfilled, onRejected, progressBack]);
-      if (this.$$state.status > 0) scheduleProcessQueue(this.$$state);
+      if (this.$$state.status > 0) scheduleProcessQueue(result.promise, this.$$state);
```

Comment:
There is something very weird with this change... why would you want the generated promise and only if the current promise is fulfill (and in this case, only some of the promises will be *marked* and some will not...